### PR TITLE
Introduce Tensor and Image archetypes for rust

### LINKED ISSUE #1992: Break up Tensor into components
The `Tensor` is currently defined as:

``` rs
pub struct Tensor {
    /// Unique identifier for the tensor
    pub tensor_id: TensorId,

    /// Dimensionality and length
    pub shape: Vec<TensorDimension>,

    /// Data payload
    pub data: TensorData,

    /// The per-element data meaning
    /// Used to indicated if the data should be interpreted as color, class_id, etc.
    pub meaning: TensorDataMeaning,

    /// Reciprocal scale of meter unit for depth images
    pub meter: Option<f32>,
}
```

`meaning` and `meter` at least should be their own components. `tensor_id` could be replaced with the `row_id` of `data`.

### PR DESCRIPTION
**Pull request by jleibs**
### What
This PR introduces the new IDL schemas for Tensors and Images.

There is a fairly significant change to how we structure our Tensor-related archetypes.

Whereas before we used a single Archetype for both Tensors and Images, they are now split into two archetypes that share a common `TensorData` component.

`TensorMeaning` is now gone and will be replaced with per-image-type archetypes: `Image`, `DepthImage`, and `SegmentationImage`

This PR only introduces the IDL and the Rust APIs.

Additional work required for: https://github.com/rerun-io/rerun/issues/2792
 - Python APIs
 - C++ APIs
 - Round-trip tests
 - DepthImage
 - SegmentationImage
 - Support for f16 types
 - View parts / heuristics / etc

Example of new Image usage:
```
    let mut image = Array::<u8, _>::zeros((200, 300, 3).f());
    image.slice_mut(s![.., .., 0]).fill(255);
    image.slice_mut(s![50..150, 50..150, 0]).fill(0);
    image.slice_mut(s![50..150, 50..150, 1]).fill(255);

    let image = Image::try_from(image)?;

    MsgSender::from_archetype("image", &image)?.send(&rec_stream)?;
```

## New Archetypes:
 - Tensor
    - TensorData
 - Image
    - TensorData

## New Components
 - TensorData
   - TensorData
 
 ## New Datatypes
 - TensorData
   - TensorId
   - TensorDimension
   - TensorBuffer

* Fixes #1992

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3032) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3032)
- [Docs preview](https://rerun.io/preview/pr%3Ajleibs%2Fhope_tensors/docs)
- [Examples preview](https://rerun.io/preview/pr%3Ajleibs%2Fhope_tensors/examples)

### CODE REVIEW

File: crates/re_types/src/datatypes/tensor_data_ext.rs
Comment (jleibs) [inline crates/re_types/src/datatypes/tensor_data_ext.rs:1]: This file is more or less copied from https://github.com/rerun-io/rerun/blob/main/crates/re_components/src/tensor.rs with small adjustments for names.

Comment (emilk) [reply crates/re_types/src/datatypes/tensor_data_ext.rs:1]: Let's add that comment to the file

**Comment by teh-cmc** [comment]
Drafting: on hold while we integrate indicator/marker components and turn each image into distinct archetypes.

**Review by teh-cmc — APPROVED** [review]
Looks great. I'll push this forward to avoid nasty conflicts with my upcoming PRs.

File: crates/re_types_builder/src/codegen/rust/serializer.rs [hunk 1]
```diff
@@ -568,7 +568,7 @@ fn quote_arrow_field_serializer(
                     .map(|datum| {
                         datum
                             .map(|datum| {
-                                let #quoted_binding = datum;
+                                let #quoted_binding = datum.clone();
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply crates/re_types_builder/src/codegen/rust/serializer.rs:571]:
```suggestion
                                let #quoted_binding = datum;

**Pull request merged**